src/include/layers/bufferedlockedheap.h

```diff
@@ -0,0 +1,108 @@
+// -*- C++ -*-
+
+#ifndef DH_BUFFEREDLOCKEDHEAP_H
+#define DH_BUFFEREDLOCKEDHEAP_H
+
+#include <mutex>
+#include <cstddef>
+
+#ifndef thread_local
```

Comment:
I have only tested it on Linux. See https://stackoverflow.com/a/18298965/3307716